yui/apps/weather/weather.py

```diff
@@ -0,0 +1,427 @@
+import asyncio
+from decimal import Decimal
+from hashlib import md5
+from typing import Optional
+from urllib.parse import urlencode
+
+import aiohttp
+from aiohttp import client_exceptions
+
+from attr import define
+
+from ...box import box
+from ...command import argument
+from ...event import Message
+from ...utils import json
+from ...utils.datetime import now
+
+box.assert_config_required("GOOGLE_API_KEY", str)
+box.assert_config_required("OPENWEATHER_API_KEY", str)
+
+
+class WeatherResponseError(RuntimeError):
+    pass
+
+
+EXCEPTIONS = (
+    client_exceptions.ClientPayloadError,  # Bad HTTP Response
+    ValueError,  # JSON Error
+    client_exceptions.ClientConnectorCertificateError,  # TLS expired
+    WeatherResponseError,  # Bad HTTP Response
+)
+
+
+LABELS = {
+    "pm25": "PM2.5",
+    "pm10": "PM10",
+    "o3": "오존",
+    "no": "일산화 질소",
+    "no2": "이산화 질소",
+    "so2": "이산화 황",
+    "co": "일산화 탄소",
+    "nh3": "암모니아",
+}
+
+
+@define
+class WeatherRecord:
+    # 위치에 대한 정보 (영어)
+    location: str
+
+    # 현재, 최고/최소 온도, 습도 및 기압, 가시거리
+    current_temp: float
+    min_temp: float
+    max_temp: float
+    feel_temp: float
+    humidity: int
+    pressure: float
+    visibility: int
+
+    # 바람의 세기와 각도
+    wind_speed: float
+    wind_degree: int
+
+    # 구름 정도
+    cloudiness: int
+
+    # 강우 유무
+    rain: bool
+    rain_1h: Optional[float]
+    rain_3h: Optional[float]
```

Comment:
Python 3.10 부터(였나...?) 다음 표현식이 가능합니다.

```suggestion
    rain_1h: float | None
    rain_3h: float | None
```

(이하 반복 생략)